### PR TITLE
Refactor BallerinaToml to replace BallerinaTomlProcessor

### PR DESCRIPTION
**Pull request by pramodya1994**
Fixes #<Issue Number>

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/test/java/io/ballerina/projects/BallerinaTomlTests.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
+package io.ballerina.projects;
```
Comment (azinneera) [inline compiler/ballerina-lang/src/test/java/io/ballerina/projects/BallerinaTomlTests.java:1]: Missing license header

Comment (pramodya1994) [reply compiler/ballerina-lang/src/test/java/io/ballerina/projects/BallerinaTomlTests.java:19]: Added

**Review by azinneera — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageManifest.java [hunk 2]
```diff
@@ -33,12 +41,12 @@
 
     // Other entries hold other key/value pairs available in the Ballerina.toml file.
     // These keys are not part of the Ballerina package specification.
-    private final Map<String, Object> otherEntries;
+    private final Map<String, TopLevelNode> otherEntries;
```
Comment (hevayo) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageManifest.java:44]: otherEntries should be a generic type and shouldn't be Toml specific. We can use Maps and List of strings to represent the data values. and keep the type as it is.

**Review by hevayo — APPROVED** [review]
Approved with some comments shall we create an issue to track it.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ProjectFiles.java [hunk 3]
```diff
@@ -147,13 +149,18 @@ private static DocumentData loadTestDocument(Path documentFilePath) {
...
+    public static PackageManifest createPackageManifest(BallerinaToml ballerinaToml) {
+        return ballerinaToml.packageManifest();
     }
 
     public static BuildOptions createBuildOptions(Path projectPath, BuildOptions theirOptions) {
         Path ballerinaTomlFilePath = projectPath.resolve(ProjectConstants.BALLERINA_TOML);
-        BuildOptions defaultBuildOptions = BallerinaTomlProcessor.parse(ballerinaTomlFilePath).getBuildOptions();
+        BallerinaToml ballerinaToml = BallerinaToml.from(ballerinaTomlFilePath);
+        if (ballerinaToml.diagnostics().hasErrors()) {
+            throw new BallerinaTomlException(ballerinaToml.getErrorMessage());
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ProjectFiles.java:160]: Should we throw the exception here? IMO, we should add it to TomlDiagnostics and continue the execution.

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/PackageConfigCreator.java [hunk 4]
```diff
@@ -53,10 +54,14 @@ public static PackageConfig createBuildProjectConfig(Path projectDirPath) {
 
         Path balTomlFilePath = projectDirPath.resolve(ProjectConstants.BALLERINA_TOML);
         BallerinaToml ballerinaToml = BallerinaToml.from(balTomlFilePath);
+
+        if (ballerinaToml.diagnostics().hasErrors()) {
+            throw new BallerinaTomlException(ballerinaToml.getErrorMessage());
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/PackageConfigCreator.java:59]: Refer https://github.com/ballerina-platform/ballerina-lang/pull/27734/files#r558386664

**Comment by pramodya1994** [comment]
> Approved with some comments shall we create an issue to track it.

Created an issue https://github.com/ballerina-platform/ballerina-lang/issues/27896

**Pull request merged**